### ISSUE TITLE
SDK 7.0.1 Too Many Crash

### ISSUE THREAD

**Issue by riskteria** [moengage/MoEngage-iOS-SDK#20]
Hi MoEngage Team,

I want to report that we have recorded several crashes from Firebase Crashlytics after updating our MoEngage SDK from version 6.2.5 to 7.0.1 which increases our app crash rate immediately. These crash behavior mostly happened when users navigate to several pages, and when our app tracks the user activity to MoEngage, the crash occurs randomly thus make it harder for us to debug it.

This issue was resolved by downgrading MoEngage iOS SDK back to version 6.2.5.

Please kindly check the SDK, why is this error occurs.

Regards

![Screen Shot 2021-03-01 at 10 16 51](https://user-images.githubusercontent.com/12659121/109633776-1648dd80-7b7b-11eb-99be-5efb7e77920f.png)

Here is some error logs

```
Crashed: com.apple.root.background-qos
0  libsystem_platform.dylib       0x1adf7e5e8 _os_unfair_lock_corruption_abort + 36
1  libsystem_platform.dylib       0x1adf7c524 _os_unfair_lock_lock_slow + 316
2  CoreFoundation                 0x1ae14da44 -[__NSArrayM dealloc] + 72
3  CoreFoundation                 0x1ae2a6040 cow_cleanup + 168
4  CoreFoundation                 0x1ae151264 -[__NSDictionaryM dealloc] + 144
5  CoreFoundation                 0x1ae2a6040 cow_cleanup + 168
6  CoreFoundation                 0x1ae151264 -[__NSDictionaryM dealloc] + 144
7  libobjc.A.dylib                0x1adfb404c AutoreleasePoolPage::releaseUntil(objc_object**) + 180
8  libobjc.A.dylib                0x1adfb3f44 objc_autoreleasePoolPop + 224
9  libdispatch.dylib              0x1adf3b504 _dispatch_last_resort_autorelease_pool_pop + 40
10 libdispatch.dylib              0x1adf217e8 _dispatch_root_queue_drain + 1036
11 libdispatch.dylib              0x1adf21cd0 _dispatch_worker_thread2 + 112
12 libsystem_pthread.dylib        0x1adf8cb38 _pthread_wqthread + 212
13 libsystem_pthread.dylib        0x1adf8f740 start_wqthread + 8
```

```
com.moengage.analytics.datatracking
0  libsystem_platform.dylib       0x1adf7db90 _platform_strcmp + 670
1  libdyld.dylib                  0x1ae07ab48 objc_opt::objc_stringhash_t::getIndex(char const*) const + 120
2  libdyld.dylib                  0x1ae07a8a8 _dyld_get_objc_selector + 60
3  libobjc.A.dylib                0x1adfac618 __sel_registerName(char const*, bool, bool) + 44
4  Foundation                     0x1ae5498ac -[NSString _stringRepresentation] + 108
5  CoreFoundation                 0x1ae15491c -[NSDictionary descriptionWithLocale:indent:] + 980
6  Foundation                     0x1ae65d960 _NSDescriptionWithLocaleFunc + 64
7  CoreFoundation                 0x1ae216cf4 __CFStringAppendFormatCore + 9516
8  CoreFoundation                 0x1ae217cbc _CFStringCreateWithFormatAndArgumentsAux2 + 152
9  Foundation                     0x1ae530814 +[NSString stringWithFormat:] + 72
10 MoEngageCore                   0x100f84a0c _hidden#957_ + 159 (__hidden#1007_:159)
11 MoEngageCore                   0x100f852d4 _hidden#961_ + 245 (__hidden#1007_:245)
12 MOAnalytics                    0x100e90abc _hidden#37_ + 529 (__hidden#292_:529)
13 libdispatch.dylib              0x1adf3a9a8 _dispatch_call_block_and_release + 24
14 libdispatch.dylib              0x1adf3b524 _dispatch_client_callout + 16
15 libdispatch.dylib              0x1adf18b3c _dispatch_lane_serial_drain$VARIANT$armv81 + 564
16 libdispatch.dylib              0x1adf1954c _dispatch_lane_invoke$VARIANT$armv81 + 396
17 libdispatch.dylib              0x1adf2284c _dispatch_workloop_worker_thread + 580
18 libsystem_pthread.dylib        0x1adf8cb74 _pthread_wqthread + 272
19 libsystem_pthread.dylib        0x1adf8f740 start_wqthread + 8
```

**Comment by ChengappaMoEngage** [comment]
We had identified one multithreading issue with Analytics Module, and have fixed the same in SDK version 7.0.3: [Release Notes](https://docs.moengage.com/docs/release-notes#v703). 
Please update to the same and let us know if are still facing any crash.

**Comment by ChengappaMoEngage** [comment]
Closing the issue for now. Let us know if still facing issues post updating the SDK.
